functions/scheduled.js

```diff
@@ -22,73 +28,69 @@ const removeDanglingSeatedUsers = async () => {
-    );
+  let removedUsersCount = 0;
+  return Promise.all(
+    chunk(recentSeatedUsers, BATCH_MAX_OPS / 2).map((userDocsBatch) => {
+      const batch = firestore.batch();
+      userDocsBatch.forEach((userDoc) => {
+        const seatedUserData = userDoc.data();
+        const userId = userDoc.id;
+        const venueId = seatedUserData.venueId;
 
-    switch (seatedUserData.template) {
-      case "auditorium":
         batch.delete(
           firestore
             .collection("venues")
             .doc(venueId)
-            .collection("sections")
-            .doc(seatedUserData.venueSpecificData.sectionId)
-            .collection("seatedSectionUsers")
+            .collection("recentSeatedUsers")
             .doc(userId)
         );
-        console.log(
-          "[scheduled-updateSeatedUsersCountInAuditorium] " +
-            `Deleted from ${seatedUserData.template} /venues/${venueId}/sections/` +
-            `${seatedUserData.venueSpecificData.sectionId}/seatedSectionUsers/${userId}`
-        );
-        break;
-      case "jazzbar":
-      case "conversationspace":
-      case "friendship":
-        batch.delete(
-          firestore
-            .collection("venues")
-            .doc(venueId)
-            .collection("seatedTableUsers")
-            .doc(userId)
-        );
-        console.log(
-          "[scheduled-updateSeatedUsersCountInAuditorium] " +
-            `Deleted from ${seatedUserData.template} /venues/${venueId}/seatedTableUsers/${userId}`
-        );
-        break;
-      default:
-        console.warn(
-          `Found unsupported venue template ${seatedUserData.template}`
-        );
-        break;
-    }
-  });
+        removedUsersCount += 1;
 
-  await batch.commit();
+        switch (seatedUserData.template) {
+          case "auditorium":
+            batch.delete(
+              firestore
+                .collection("venues")
+                .doc(venueId)
+                .collection("sections")
+                .doc(seatedUserData.venueSpecificData.sectionId)
+                .collection("seatedSectionUsers")
+                .doc(userId)
+            );
+            break;
+          case "jazzbar":
+          case "conversationspace":
+            batch.delete(
+              firestore
+                .collection("venues")
+                .doc(venueId)
+                .collection("seatedTableUsers")
+                .doc(userId)
+            );
+            break;
+          default:
+            console.warn(
+              `Found unsupported venue template ${seatedUserData.template}`
+            );
+            break;
+        }
+      });
+      return batch.commit();
+    })
+  ).then(() => console.log(`Removed ${removedUsersCount} dangling users`));
 };
 
-exports.updateSeatedUsersCountInAuditorium = functions.pubsub
-  .schedule("every 5 minutes")
-  .onRun(async () => {
+exports.updateSeatedUsersCountInAuditorium =
+  // functions.https.onCall(async () => {
```

Comment:
remove?

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/2361/commits/81f321955aeab6f1da1596d30eb3fd572a53fb14